### PR TITLE
Feature crear wyelab 159979848

### PR DESCRIPTION
**Pull request by Gimli76**
Added model, endpoint, factory and specs for creating Wyelabs.

Not much to mention. Validations include:
- Title must be present, and no longer than 100 characters
- Date must be present, and must be in the future (at creation time)
- Speakers must be present

### CODE REVIEW

File: spec/factories/wyelabs.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    trait :valid do
+      title { FFaker::Book.title }
+      lab_date { FFaker::Time.between(Time.zone.now + 5.seconds, Time.zone.now + 1.year) }
+      speakers { FFaker::Name.name + " " + FFaker::Name.name }
+      description { FFaker::Lorem.paragraph }
+    end
+
+    trait :long_title do
+      title { FFaker::Lorem.characters(101) }
+      lab_date { FFaker::Time.between(Time.zone.now + 5.seconds, Time.zone.now + 1.year) }
```
Comment (jpducassou) [inline spec/factories/wyelabs.rb]: I think you should not repeat attribute's generation code on traits and only override not inherited ones.

Comment (Gimli76) [reply spec/factories/wyelabs.rb]: Good catch!

File: app/controllers/wyelabs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class WyelabsController < ApplicationController
+
+  # POST /wyelabs
+  def create
+    Wyelab.transaction do
```
Comment (joaquinvillar) [inline app/controllers/wyelabs_controller.rb]: why do you need a transaction if you have one action?

File: spec/controllers/wyelab_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      it "creates a new Wyelab" do
+        allow(controller).to receive(:authorize)
+        post :create, params: no_title
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "when title is too long" do
+      let(:no_title) do
```
Comment (joaquinvillar) [inline spec/controllers/wyelab_controller_spec.rb]: maybe this have to be named as long_title

File: spec/controllers/wyelab_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      it "creates a new Wyelab" do
+        allow(controller).to receive(:authorize)
+        post :create, params: no_title
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "when lab_date is missing" do
+      let(:no_title) do
```
Comment (joaquinvillar) [inline spec/controllers/wyelab_controller_spec.rb]: maybe this have to be named as no_lab_date

File: spec/controllers/wyelab_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      it "creates a new Wyelab" do
+        allow(controller).to receive(:authorize)
+        post :create, params: no_title
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "when speakers is missing" do
+      let(:no_title) do
```
Comment (joaquinvillar) [inline spec/controllers/wyelab_controller_spec.rb]: maybe this have to be named as no_speakers

File: spec/controllers/wyelab_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      it "creates a new Wyelab" do
+        allow(controller).to receive(:authorize)
+        post :create, params: no_title
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "when date is from the past" do
+      let(:no_title) do
```
Comment (joaquinvillar) [inline spec/controllers/wyelab_controller_spec.rb]: maybe this have to be named as past_date

File: app/controllers/wyelabs_controller.rb (hunk 2 above)
Comment (Gimli76) [reply app/controllers/wyelabs_controller.rb]: Good question my friend, and good review by the way. Thanks!

**Pull request merged**